### PR TITLE
feat: add proxy service for simplicity of extending service chain

### PR DESCRIPTION
**Pull request by wtrocki**
Idea here is to simplify creation of the service chains even further. 
Currently you need to creat entire class witch is very verbose. 
Adding proxy service to the core and extending it is very simple and can be reused to build long chain of services without implementing entire API

### CODE REVIEW

File: packages/graphback-core/src/runtime/index.ts [hunk 1]
```diff
@@ -4,4 +4,5 @@ export * from './GraphbackDataProvider'
 
 export * from "./CRUDService"
 export * from "./createCRUDService"
-export * from "./NoDataError"
\ No newline at end of file
+export * from "./NoDataError"
+export * from "./GraphbackProxyService"
```
Comment (machi1990) [inline packages/graphback-core/src/runtime/index.ts:8]: Hmm where does the ProxyService class come from?

Comment (machi1990) [reply packages/graphback-core/src/runtime/index.ts:8]: CI failure https://app.circleci.com/pipelines/github/aerogear/graphback/5195/workflows/47abd466-58b7-4997-9920-616de11ad610/jobs/5296/steps related.

Comment (wtrocki) [reply packages/graphback-core/src/runtime/index.ts:8]: Yep.. It wasn't staged. My bad

File: packages/graphback-core/src/runtime/GraphbackProxyService.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  public update(data: T, context: any): Promise<T> {
+    return this.proxiedService.update(data, context);
+  }
+
+  public delete(data: T, context: any): Promise<T> {
+    return this.proxiedService.delete(data, context);
+  }
+
+  public findOne(args: any, context: any): Promise<T> {
+
```
Comment (machi1990) [inline packages/graphback-core/src/runtime/GraphbackProxyService.ts:31]:
```suggestion
```

File: packages/graphback-core/src/runtime/GraphbackProxyService.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
+import { GraphbackCRUDService, ResultList } from './GraphbackCRUDService';
+import { GraphbackOrderBy, GraphbackPage } from '.';
+
+/**
+ * This custom CRUD Service shows another potential way to add auth
+ *
+ * This is actually quite nice and clean but it does not allow for field level auth.
+ * It's still a possibility that we could go with though!
+ */
```
Comment (machi1990) [inline packages/graphback-core/src/runtime/GraphbackProxyService.ts:9]: Comments needs update

File: packages/graphback-core/src/runtime/GraphbackProxyService.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
```
Comment (wtrocki) [inline packages/graphback-core/src/runtime/GraphbackProxyService.ts:9]:
```suggestion
/**
 * ProxyService that can be used by any services that wish to extend 
 * Graphback functionality.
 * Service works by proxying method requests to another service or 
 * datastore.
 */
```

**Comment by wtrocki** [comment]
Note I do not want to document this yet. Going to try this approach with multiple services and see if that will be useful. This stuff is more for us as maintainers to be able to build another set of services like AMQ and build some wrappers etc.

**Pull request merged**